services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -476,6 +482,16 @@ private void handleD2ServiceLookup(ChannelHandlerContext ctx, VenicePathParserHe
       setupErrorD2DiscoveryResponseAndFlush(NOT_FOUND, errorMsg, ctx);
       return;
     }
+
+    // Only create metrics for valid stores
+    Sensor d2DiscoverySensor = metricsRepository.sensor("venice.router.d2_discovery." + storeName);
```

Comment:
Let's follow existing patterns and make a new class that extends `AbstractVeniceStats` and use the register sensor methods there instead of calling `metricsRepository.sensor` directly and performing `metricsRepository.metrics().containsKey`. See example usages of `AbstractVeniceStats` and `AbstractVeniceAggStats` for store level stats.